docs/rfc/rfc-19-random-beacon-v2.adoc

```diff
@@ -0,0 +1,225 @@
+giving about 1.5 hours to submit a relay entry before the minimum stake from each
+group member was slashed. This timeout proved to be not enough to notify,
+investigate, and fix a problem. Three groups were slashed in a row during the
+November 11th incident that begun around 08:00 UTC, before any US-based
+operators noticed the problem.
+
+
+== Proposal
+
+=== Group creation
+
+New groups should be created with a fixed, governable frequency of relay
+requests. Instead of a ticket-based approach for group selection, we should
+use sortition pools. Group creation start transaction should be embedded into
+relay request transaction and should lock a sortition pool. From this moment,
+no operator can enter or leave the pool. All off-chain clients should perform
+group selection optimistically, using a view sortition function call and perform
+DKG. Once the group submits the result to the chain, a challenge period starts.
+
+During the challenge period, anyone can notify that the submitted DKG result
+contains group members not selected by the pool. This leads to slashing all
+members who signed the result and the notifier receives 5% from the slashed amount.
+The length of the challenge period and slashing amount are governable parameters.
+
+Once the challenge period passes, anyone can unlock the sortition pool and mark
+the DKG result as accepted. In this transaction, the DKG result submitter receives
+a reward, as described in <<fees-and-rewards, Fees and Rewards>> section.
+
+In case the DKG result was not submitted before a timeout, anyone can notify DKG
+timed out and receive a reward, as described in <<fees-and-rewards, Fees and Rewards>> 
+section.
+
+The sortition pool should weigh operators by stake and allow to select the same 
+operator multiple times. DKG protocol should execute in the same way as for v1
+and inactive/disqualified members should be marked and kicked out of the
+sortition pool for at least 2 weeks losing their rewards for that time.
+
+[[fees-and-rewards]]
+=== Fees and rewards
+
+In tBTC v1, operators have to execute a lot of on-chain transactions per
+deposit. Even though they are not reimbursed for on-chain actions and they
+receive KEEP rewards and TBTC fees, this scheme proved to be successful and no
+operators complained about it.
+
+For v2 of the random beacon, all rewards should be done in T.
+This scheme is compatible with tBTC v2 design where all rewards and
+are also done only in T.
+
+Relay requester should provide a fee in T. The value of the fee is a governable
+parameter. The entire fee is deposited in the maintenance pool that is used to
+reimburse for different actions related to DKG.
+
+The transaction submitting relay entry is not reimbursable and implementation
+should ensure the gas cost of this transaction is as low as possible, below
+200k gas. The first group member eligible to submit the result is 
+`previous_entry MOD group_size`, then also `(previous_entry MOD group_size) + 1`.
```

Comment:
why not `new_entry % group_size`? this way it would be unpredictable so there would be no safe way to be inactive

Reply:
Hmm... yes, even better!

Reply:
I really like it!

Reply:
Please see https://github.com/keep-network/keep-core/pull/2585/commits/5ca9187cf42019592e814451eb2d8935c9d41dde.